### ISSUE TITLE
User prompt breaks usability, please refactor the option switch

### ISSUE THREAD

**Issue by shyd** [rarcega/instagram-scraper#163]
Hi,

I just updated to v1.5.19. I like the user prompt implemented by @VasiliPupkin256 in #158 
But this breaks the ability to fire and forget the scraping, as it prompts every single time media is affected by #157.
Enabling the switch `--retry-forever` doesn't help, because all threads run into endless loops retrying but always getting 403. (during my scrape there are multiple files with 403)

Please refactor this to let's say `--retry-policy=[abort,ignore,retry]`. Well, `abort` seems to be duplicate for `ignore`.

I love this tool, but I don't feel that comfortable with the recent update.

Cheers 🍻

**Comment by VasiliPupkin256** [comment]
Without a prompt it just crash. How a prompt break usability? #157 breaks usability.
Ignoring all the 403 errors is not a solution and shouldn't be implemented. The function `get_original_image` should be changed, or the other version of url downloaded in case of 403 error.
Also abort is only duplicate to ignore on a separate media file, used in main loop it stops execution of a program in most cases.

**Comment by shyd** [comment]
I agree on `get_original_image`. But then there should be an option like `--fallback-to-non-original-image` to prevent the 403s.

But for now I cannot start an automated run via e.g. cron due to the prompts during the run.

But if I could set a default answer to all these prompts, it would run like it did before the update.

I think the bottle is half full with only the `--retry-forever` option.

Please don't get me wrong, as I really appreciate your work! 👍

**Comment by VasiliPupkin256** [comment]
`--retry-forever` address the other issues, temporal bans, #139, #149 etc. 
403 response code is rather simple case and doesn't require any special command line switch

**Comment by shyd** [comment]
Totally agree.

But do you get my point about automated runs?

**Comment by VasiliPupkin256** [comment]
I did. I would disable the prompt if `--quiet` switch is set

**Comment by shyd** [comment]
Hm this way I won't be able to take a look at the output after the automated run. 

Why not:

- add `--retry-policy=[abort,ignore,retry]`

- or enable the prompt with `--interactive-mode`?

**Comment by VasiliPupkin256** [comment]
Well the other option is to utilise `interactive` switch as an opposite to `quiet`, I just don't want to change command line interface much for a temporal fix

**Comment by shyd** [comment]
I like the `interactive` switch, because it preserves the progress output if it's not set 👍

**Comment by VasiliPupkin256** [comment]
Using the `interactive` switch was bad idea because prompt should be the default option, seems like the "ignore all" behavior can't be implemented without adding additional switches. But I've found an easy fix to #157 so the issue is not as acute now

**Comment by shyd** [comment]
I've seen your PR already. Thanks for the fix!
I can confirm, that stripping the url further eliminated the 403s 👍

**Comment by genvarek** [comment]
Mode without prompt still needed for automated work! It would be super great if the scraper could ignore errors like "Repeated error HTTPError" and continue by itself
